### PR TITLE
Fixed helm uninstall not deleting the resource.

### LINKED ISSUE #10588: Helm uninstall command deletes the release but does not delete resources in case of network issues
Output of `helm version`:

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

### PR DESCRIPTION
**Pull request by IamMayankThakur**
Signed-off-by: Mayank Thakur <thakurmayank88@gmail.com>

**What this PR does / why we need it**: Added `nil` check for `errs` when deleting the release. Closes #10588 . 

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by joejulian — APPROVED** [review]
lgtm, but I am not a core maintainer. Still needs an approval from a core maintainer.

**Review by bacongobbler — APPROVED** [review]
LGTM.

**Pull request merged**
